sdcm/stress_thread.py

```diff
@@ -135,11 +135,21 @@ def create_stress_cmd(self, node, loader_idx, keyspace_idx):
         return stress_cmd
 
     @staticmethod
-    def _add_errors_option(stress_cmd: str, to_add: list):
-        to_add = (' '.join(to_add))
-        if ' -errors' not in stress_cmd:
-            return stress_cmd + ' -errors ' + to_add
-        return re.sub('-errors([ ]+[^-][a-zA-Z0-9-]+)+([ ]* -[a-z]+|[ ]*)', '-errors\\1 ' + to_add + '\\2', stress_cmd)
+    def _add_errors_option(stress_cmd: str, to_add: list) -> str:
+        """
+        Add suboption to -errors option, if such suboption is there, does not add or change it
+        """
+        current_options = stress_cmd.split(' -')
+        to_add = list(to_add)
+        current_error_option = next((option for option in current_options if option.startswith('errors ')), None)
+        if current_error_option is None:
+            return stress_cmd + ' -errors ' + ' '.join(to_add)
+        to_add_heads = [suboption.split('=')[0] for suboption in to_add]
+        suboption_heads = [suboption.split('=')[0] for suboption in current_error_option.split(' ')[1:]]
+        new_to_add = [option for option, head in zip(to_add, to_add_heads) if head not in suboption_heads]
+        if not new_to_add:
+            return stress_cmd
+        return stress_cmd.replace(current_error_option, current_error_option + ' ' + ' '.join(new_to_add))
```

Comment:
It's little bit over-complicated.

There two things which can be done simpler:

1. Extracting current `-errors` option:

```python
# Split whole command by ` -errors `. Btw, we can check the command for duplicates here too. If we want, 'cause.
head, *errors_opts = stress_cmd.split(" -errors ", 1)
tail = ""

if errors_opts:  # This list is empty if there is no `-error' option in the `stress_cmd'
    errors_opts, *tail = errors_opts[0].split(" -", 1)
    errors_opts = errors_opts.split()
    tail = f" -{tail[0]}" if tail else ""
```    

So, you have string `head`, list of errors options `errors_opts`, and string `tail` now.

2. Another simplification can be applied there is using `dict` for collecting required opts to add:

```python
new_errors_opts = list({opt.split("=", 1)[0]: opt for opt in list(to_add) + errors_opts}.values())
```

If we really need to return original `stress_cmd` in case we don't add any errors options then we can compare lengths of `errors_opts` and `new_errors_opts`. But we can skip it and just return result:

```python
return f"{head} -errors {' '.join(new_errors_opts)}{tail}"
```